impl/client/src/main/javascript/web/util/formatting.js

```diff
@@ -192,6 +192,30 @@ define("common-ui/util/formatting", ['common-ui/util/timeutil', 'common-ui/util/
           }
 
           if(timezone == 'client') {
+
```

Comment:
So are you handling case 4?
> 4. customized timezone - timezone forced to customized timezone

Reply:
No, in this code snippet I am handling case 2 (i.e. when the display is based on client timezone).
Regarding case 4, it's only handled in server side and it was implemented in this same delivery but in the server's repository, please check: https://github.com/pentaho/pentaho-platform-plugin-reporting/pull/627